File: pygraph/graph/state.py
from typing import TypedDict, List, Dict, Any, Optional


class CompareState(TypedDict):
    user_request: str
    category: str
    parsed_preferences: Dict[str, Any]
    candidates: List[Dict[str, Any]]
    shortlisted: List[Dict[str, Any]]
    recommendation: Optional[str]

File: server/graph/compare/state.js
const { Annotation } = require("@langchain/langgraph");

const CompareState = Annotation.Root({
  user_request: Annotation({
    value: (left, right) => right ?? left,
    default: () => "",
  }),
  category: Annotation({
    value: (left, right) => right ?? left,
    default: () => "",
  }),
  parsed_preferences: Annotation({
    value: (left, right) => right ?? left,
    default: () => ({}),
  }),
  candidates: Annotation({
    value: (left, right) => right ?? left,
    default: () => [],
  }),
  shortlisted: Annotation({
    value: (left, right) => right ?? left,
    default: () => [],
  }),
  recommendation: Annotation({
    value: (left, right) => right ?? left,
    default: () => null,
  }),
});

module.exports = {
  CompareState,
};
